Sources/Maintenance/Migration/v3_0/UserReactions.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines https://www.simplemachines.org
+ * @copyright 2024 Simple Machines and individual contributors
+ * @license https://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 3.0 Alpha 3
+ */
+
+declare(strict_types=1);
+
+namespace SMF\Maintenance\Migration\v3_0;
+
+use SMF\Db\DatabaseApi as Db;
+use SMF\Maintenance\Migration\MigrationBase;
+use SMF\Db\Schema\v3_0;
+use SMF\Db\Schema\v3_0\UserReacts;
+
+class UserReactions extends MigrationBase
+{
+	/*******************
+	 * Public properties
+	 *******************/
+
+	/**
+	 *
+	 */
+	public string $name = 'Converting likes to reactions';
+
+	/****************
+	 * Public methods
+	 ****************/
+
```

Comment:
```suggestion
	public function isCandidate(): bool
	{
		// Make sure we haven't already done this...
		$cols = Db::$db->list_columns('messages');

		// If the reactions column exists in the messages table, there's nothing to do
		return !in_array('reactions', $cols))
    }

```

If you add this check, you can remove it from the execute() function.  The upgrader will indicate "skip" when it comes across it and isCandidate() returns false.